src/System.Management.Automation/DscSupport/CimDSCParser.cs

```diff
@@ -3261,14 +3261,15 @@ public static bool ImportCimKeywordsFromModule(PSModuleInfo module, string resou
                 //
                 try
                 {
-                    var dscResourceDirectories = Directory.GetDirectories(dscResourcesPath);
-                    foreach (var directory in dscResourceDirectories)
+                    foreach (var directory in Directory.EnumerateDirectories(dscResourcesPath))
                     {
-                        var schemaFiles = Directory.GetFiles(directory, "*.schema.mof", SearchOption.TopDirectoryOnly);
-                        if (schemaFiles.Length > 0)
+                        var schemaFiles = Directory.EnumerateFiles(directory, "*.schema.mof", SearchOption.TopDirectoryOnly);
+
+                        Debug.Assert(!schemaFiles.Skip(1).Any(), "A valid DSCResource module can have only one schema mof file");
```

Comment:
Is this the right logic?  Or am I misreading it?

Reply:
Yep, I was trying to be clever to avoid enumerating the entire sequence with `Count`, like:

```csharp
Debug.Assert(schemaFiles.Count() == 1, "A valid DSCResource module can have only one schema mof file");
```

There is in fact a subtle difference in logic because `Skip` doesn't throw when the source is empty.

Reply:
I've also introduced a multiple enumeration here, even if it is only in debug build.

Reply:
In the case the assert should be after 3270 line.

Reply:
That should do the trick!